borg/testsuite/archiver.py

```diff
@@ -1122,8 +1129,9 @@ def setUp(self):
 
     def open_archive(self, name):
         repository = Repository(self.repository_path)
-        manifest, key = Manifest.load(repository)
-        archive = Archive(repository, key, manifest, name)
+        with repository:
+            manifest, key = Manifest.load(repository)
+            archive = Archive(repository, key, manifest, name)
         return archive, repository
```

Comment:
hmm, if you use "with repository" above, it will return a closed repository here.